src/Task.ts

```diff
@@ -289,9 +290,10 @@ export class Task {
         // Tags are found in the string and pulled out but not removed,
         // so when returning the entire task it will match what the user
         // entered.
+        // The global filter will be removed from the collection.
         const hashTagMatch = description.match(this.hashTags);
         if (hashTagMatch !== null) {
-            tags = hashTagMatch;
+            tags = hashTagMatch.filter((tag) => tag !== globalFilter);
```

Comment:
Interesting, removal happens when tasks updates/ displays the task. In the processing the filter will actually be there and match so be in the collection. 

To fix this the process that tasks uses to remove the filter needs to be reviewed. I can fix it by looking for the remove setting and handle that in the query.

Reply:
> Interesting, removal happens when tasks updates/ displays the task. In the processing the filter will actually be there and match so be in the collection.

Ah right, so that explains the behaviour now with the search for tags. Thank you.

> To fix this the process that tasks uses to remove the filter needs to be reviewed. I can fix it by looking for the remove setting and handle that in the query.

It could be done by removing it in the query. It could also be done by remove it from the stored tags, and the advantage of that is that it removes the need for every bit in the code that uses the tags to have to check the settings.

Reply:
I also think it's fine to remove the global filter from the list of parsed tags. The question remains what should happen with sub-tags of the global tag, right? Even though that was never officially supported :sweat_smile: 

All global filter matches are currently removed. That's probably not good, but it should be consistent for now.

Would it make sense to remove all tags that match the global tag? @claremacrae what do you think?